### PR TITLE
Allow cases to be re-assigned after addition of participant model

### PR DESCRIPTION
**Pull request by wssheldon**
Still testing this, wanted to push changes in case @kevgliss is working on it.

### CODE REVIEW

File: src/dispatch/case/service.py [hunk 1]
```diff
@@ -230,23 +230,24 @@ def update(*, db_session, case: Case, case_in: CaseUpdate, current_user: Dispatc
...
+            participant_flows.remove_participant(
+                case.assignee.individual.email,
+                case,
+                db_session,
+            )
+            event_service.log_case_event(
+                db_session=db_session,
+                source="Dispatch Core App",
+                description=f"Case assigned to {case_in.assignee.individual.email} by {current_user.email}",
+                dispatch_user_id=case.assignee.id,
```
Comment (wssheldon) [inline src/dispatch/case/service.py:248]:
```suggestion
                dispatch_user_id=current_user.id,
```

File: src/dispatch/participant/service.py [hunk 2]
```diff
@@ -44,6 +44,17 @@ def get_by_incident_id_and_email(
     )
 
 
+def get_by_case_id_and_email(*, db_session, case_id: int, email: str) -> Optional[Participant]:
+    """Get a participant by incident id and email."""
```
Comment (wssheldon) [inline src/dispatch/participant/service.py:48]:
```suggestion
    """Get a participant by case id and email."""
```

File: src/dispatch/case/service.py [hunk 3]
```diff
@@ -230,23 +230,28 @@ def update(*, db_session, case: Case, case_in: CaseUpdate, current_user: Dispatc
 
     if case_in.assignee:
         if case.assignee.individual.email != case_in.assignee.individual.email:
-            case_assignee = auth_service.get_by_email(
-                db_session=db_session, email=case_in.assignee.individual.email
+            # We change the current assignee role to participant
+            role_flows.change_role_flow(
+                db_session=db_session,
+                case=case,
```
Comment (wssheldon) [inline src/dispatch/case/service.py:236]:
```suggestion
                subject=case,
```

File: src/dispatch/case/flows.py [hunk 4]
```diff
@@ -639,3 +650,21 @@ def case_to_incident_endpoint_escalate_flow(
...
+
+def case_assign_role_flow(
+    case_id: int,
+    assignee_email: str,
+    assignee_role: str,
+    db_session: SessionLocal,
+):
+    """Runs the case participant role assignment flow."""
+
+    from dispatch.participant_role import flows as role_flow
```
Comment (kevgliss) [inline src/dispatch/case/flows.py:663]: Is this a circular import issue?

**Review by kevgliss — APPROVED** [review]

File: src/dispatch/case/flows.py (hunk 4 above)
Comment (wssheldon) [reply src/dispatch/case/flows.py]: Nop, just a typo, moved to top of file.

**Pull request merged**